### PR TITLE
bugfix: fix containerd pid may be reuse

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

### Ⅰ. Describe what this PR did
In case containerd process may still alive on the host when the pouchd starting, we cache the containerd pid in a file `container.pid` and the pid file will be deleted when the pouchd exited.

In this case, we can check if the pid of containerd still alive, if true pouchd will not run a new containerd instance, but just take over the old containerd instance.

But there has a situation that the pid file cannot be deleted that the host was broken down by accident, and when the host restart again the pid file record a pid reused by the other process but the containerd. So when the pouchd restarting, it will check the containerd is alived but cannot take over the process because the process is not the containerd. 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

add PouchDaemonSuite.TestContainerdPIDReuse

### CODE REVIEW

File: ctrd/supervisord/daemon.go [hunk 1]
```diff
@@ -167,6 +167,29 @@ func (d *Daemon) healthPostCheck() error {
...
 }
 
+func (d *Daemon) isContainerdProcess(pid int) (bool, error) {
+	if !utils.IsProcessAlive(pid) {
+		return false, nil
+	}
+
+	// get process path by pid, if readlink -f command exit code not equals 0,
+	// we can ensure the pid is not owned by containerd.
+	output, err := exec.Command("readlink", "-f", fmt.Sprintf("/proc/%d/exe", pid)).Output()
```
Comment (Ace-Tang) [inline ctrd/supervisord/daemon.go]: Can we use `os.Readlink()`

File: ctrd/supervisord/daemon.go [hunk 2]
```diff
@@ -167,6 +167,29 @@ func (d *Daemon) healthPostCheck() error {
...
+	if err != nil {
+		return false, nil
+	}
+	processPath := strings.TrimSpace(string(output))
+
+	// get containerd process path
+	output, err = exec.Command("which", d.binaryName).Output()
```
Comment (Ace-Tang) [inline ctrd/supervisord/daemon.go]: Use `exec.LookPath()` ?

**Comment by rudyfly** [comment]
LGTM

**Comment by rudyfly** [comment]
LGTM

**Pull request merged**